torch/csrc/jit/frontend/environment.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+#include <torch/csrc/jit/frontend/ir_emitter_utils.h>
+
+namespace torch {
+namespace jit {
+
+using ValueTable = std::unordered_map<std::string, SugaredValuePtr>;
+using TypeTable = std::unordered_map<std::string, TypePtr>;
+
+// Auxiliary data structure for desugaring variable binding into our always
+// explicitly scoped language as we descend down nested control structures in
+// the frontend (which themselves don't introduce scopes)
+//
+// The Environment keeps track of two tables, one for values which are not first
+// class and a type table for values which are. When a first class value
+// is set in the environment, we emit a prim::Store which sets the
+// name of the variable to appropriate type, and when a first-class value is
+// referenced we emit a prim::Load that generates a value of the appropriate
+// type.
+//
+// a = 1
+// print(a)
+// becomes:
+// = prim::Store[name="a"](%a.1)
+// %a : int = prim::Load[name="a"]()
+// prim::Print(%a)
+struct Environment {
+  Environment(
+      Function& method,
+      ResolverPtr resolver,
+      Block* b,
+      std::shared_ptr<Environment> next = nullptr)
+      : method(method),
+        resolver(std::move(resolver)),
+        b(b),
+        next(std::move(next)) {}
+
+  // NOLINTNEXTLINE(cppcoreguidelines-non-private-member-variables-in-classes)
+  Function& method;
+  // NOLINTNEXTLINE(cppcoreguidelines-non-private-member-variables-in-classes)
+  ResolverPtr resolver;
+  // NOLINTNEXTLINE(cppcoreguidelines-non-private-member-variables-in-classes)
+  std::unordered_map<std::string, std::function<std::string()>> error_messages;
+  // NOLINTNEXTLINE(cppcoreguidelines-non-private-member-variables-in-classes)
+  Block* b;
+
+  // NOLINTNEXTLINE(cppcoreguidelines-non-private-member-variables-in-classes)
+  std::shared_ptr<Environment> next;
```

Comment:
It is generally a bad code smell if lint warnings are suppressed.

Maybe you can try following approaches to improve code quality
1) Make them private or protected depending on access pattern they need.
2) Add getter methods if access to these fields are indeed needed
3) Declare classes, that need direct access to these fields, as friend classes

Reply:
Not my code; I just copied this and moved it to a different file. Perhaps a follow-up PR could be deleting lint warnings? Or should I do it in this PR?

Reply:
I went ahead and deleted them even though it's not my code. Let's consider it part of the clean-up effort :)